### ISSUE TITLE
Kotlin support

### ISSUE THREAD

**Issue by morgante** [getgrit/gritql#570]
We would like to support Kotlin as a language for matching against.

Review [this guide](https://github.com/getgrit/gritql/blob/main/CONTRIBUTING.md#adding-a-new-target-language) and join us on Discord to discuss.

## Acceptance criteria
1. Working grammar + language support
2. At least 10 test cases, including rewrites and metavariables.

**Comment by Alex-ley-scrub** [comment]
@urbit-pilled yeah I’ll be completing it. I’ve been chatting to @morgante on Discord. I was waiting for some tips from him regarding where to look. Essentially, I’ve done ~90+% of the work and it looks like the metavariable grammar is working fine etc. but then some snippets don’t match (hence broken tests - the tests I added to make sure the snippets where matching correctly). He’s told me where to have a look now so I can restart the work.

**Comment by urbit-pilled** [comment]
Ok, good to hear @Alex-ley-scrub. If you need help you can message me on discord. 

In my experience, this task is more complex than it initially appears. A few months ago, I add PHP and Ruby to Grit for a much larger amount, and I initially underestimated the difficulty of the task. As I began working on it, I encountered numerous problems along the way. 

You can take a look at [my fork](https://github.com/urbit-pilled/gritql/tree/add_kotlin) if you want to see a kotlin branch that passes a few test cases.
